alert.go

```diff
@@ -558,7 +571,10 @@ func (a *AlertNode) runAlert([]byte) error {
 			var highestPoint *models.BatchPoint
 
 			var currentLevel alert.Level
-			if state, ok := a.states[b.Group]; ok {
+			a.cardinalityMu.RLock()
```

Comment:
I think this pattern is pretty common throughout, should I add it to the other nodes that have this pattern as well?